File: app/src/main/java/com/mapelli/simone/githubclient/viewmodel/UserRepositoriesViewModelFactory.java
package com.mapelli.simone.githubclient.viewmodel;


import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class UserRepositoriesViewModelFactory extends ViewModelProvider.NewInstanceFactory {
    private final static String TAG = UserRepositoriesViewModelFactory.class.getSimpleName();

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new UserRepositoriesViewModel();
    }
}

File: app/src/main/java/com/mapelli/simone/githubclient/viewmodel/UserRepositoriesViewModelFactory.kt
package com.mapelli.simone.githubclient.viewmodel


import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class UserRepositoriesViewModelFactory : ViewModelProvider.NewInstanceFactory() {

    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return UserRepositoriesViewModel() as T
    }

    companion object {
        private val TAG = UserRepositoriesViewModelFactory::class.java!!.getSimpleName()
    }
}
